doc/source/serve/deploy_model_tutorial.rst

```diff
@@ -0,0 +1,315 @@
+
+    input_ids = tokenizer.encode(text, return_tensors="pt", add_special_tokens=True)
+    generated_ids = model.generate(input_ids=input_ids, num_beams=2, max_length=max_length,  repetition_penalty=2.5, length_penalty=1.0, early_stopping=True)
+    preds = [tokenizer.decode(g, skip_special_tokens=True, clean_up_tokenization_spaces=True) for g in generated_ids]
+    
+    return preds[0]
+  
+  article_text = ("HOUSTON -- Men have landed and walked on the moon. "        
+  "Two Americans, astronauts of Apollo 11, steered their fragile "            
+  "four-legged lunar module safely and smoothly to the historic landing "     
+  "yesterday at 4:17:40 P.M., Eastern daylight time. Neil A. Armstrong, the " 
+  "38-year-old commander, radioed to earth and the mission control room "     
+  "here: \"Houston, Tranquility Base here. The Eagle has landed.\" The "      
+  "first men to reach the moon -- Armstrong and his co-pilot, Col. Edwin E. " 
+  "Aldrin Jr. of the Air Force -- brought their ship to rest on a level, "    
+  "rock-strewn plain near the southwestern shore of the arid Sea of "         
+  "Tranquility. About six and a half hours later, Armstrong opened the "      
+  "landing craft\'s hatch, stepped slowly down the ladder and declared as "   
+  "he planted the first human footprint on the lunar crust: \"That\'s one "   
+  "small step for man, one giant leap for mankind.\" His first step on the "  
+  "moon came at 10:56:20 P.M., as a television camera outside the craft "     
+  "transmitted his every move to an awed and excited audience of hundreds "   
+  "of millions of people on earth.")
+  
+  article_summary = summarize(article_text, 100)
+  
+  print(article_summary)
+
+  '''
+  Print output:
+
+  and walked on the moon. The astronauts of Apollo 11 steered their craft to
+  the historic landing yesterday at 4:17:40 P.M., Eastern daylight time.
+  "That's one small step for man, one giant leap for mankind," Armstrong
+  declared as he planted the first human footprint on the lunar crust. He
+  radioed to earth and the mission control room: "Houston, Tranquility Base
+  here. The Eagle has landed."
+  '''
+
+The ``tokenizer`` and ``model`` variables store our model's input encodings and weights.
+The ``summarize`` function takes in article ``text`` and the summary's ``max_length``, and then it summarizes the article text.
+In the example above, the ``summarize`` function generates a concise version of a lengthy news snippet about the moon landing.
+
+The goal is to deploy this model using Ray Serve, so it can be queried over HTTP.
+We'll start by converting the above Python function into a Ray Serve deployment that can be run locally on a laptop.
+
+First, we need to import ``ray``, ``ray serve``, and ``requests``.
+The ``ray`` and ``ray serve`` libraries give us access to Ray Serve's deployments, 
+so we can access our model over HTTP. The requests library handles HTTP request routing:
+
+.. code-block:: python
```

Comment:
use literalinclude to make the script testable https://github.com/ray-project/ray/blob/master/doc/source/serve/pipeline.rst#basic-api